crates/sui-adapter-transactional-tests/tests/receive_object/pt_receive_type_fixing.move

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//# init --addresses tto=0x0
+
+//# publish
+module tto::M1 {
+    use sui::object::{Self, UID};
+    use sui::tx_context::{Self, TxContext};
+    use sui::transfer::{Self, Receiving};
+    use std::vector;
+
+    struct A has key, store {
+        id: UID,
+    }
+
+    struct B has key, store {
+        id: UID,
+    }
+
+    public fun start(ctx: &mut TxContext) {
+        let a = A { id: object::new(ctx) };
+        let a_address = object::id_address(&a);
+        let b = B { id: object::new(ctx) };
+        transfer::public_transfer(a, tx_context::sender(ctx));
+        transfer::public_transfer(b, a_address);
+    }
+
+    public entry fun pass_through(x: Receiving<B>): Receiving<B> { x }
+
+    public entry fun pass_through_a(x: Receiving<A>): Receiving<A> { x }
+
+    public entry fun pass_through_ref_a(_x: &Receiving<A>) { }
+
+    public entry fun pass_through_mut_ref_a(_x: &mut Receiving<A>) { }
```

Comment:
Did we end up fixing the type on first usage? Or first mutable usage?
If the latter, I have some more tests for you :)

Reply:
We did windup fixing the type on first usage!